### DISCUSSION TITLE
Why do source generators still require `netstandard2.0`?

### DISCUSSION THREAD

**Discussion by colejohnson66** [dotnet/roslyn#72777] [General]
Roslyn source generators are stuck at `netstandard2.0`. Is there a reason why? We are missing modern APIs that require adding NuGet package references and writing logic to bundle them into the `nupkg`. Record types also can't be used unless one adds a polyfill for `System.Runtime.CompilerServices.IsExternalInit`. Each year, the set of usable APIs grows further and further apart from what we should have.

Are there any plans to address these limitations? If not, why?

**Comment by chrisoverzero** [comment]
> Is there a reason why?

They need to be called from .NET (Core) and .NET Framework contexts. Visual Studio is still, in some way, built atop .NET Framework, so source generators must remain compatible.

**Comment by huoyaoyuan** [comment]
This isn't limited to source generator. Roslyn itself still need to target netstandard2.0 to support VS.
Some team member said "nothing to share" about moving VS to .NET Core earlier, but we can infer something is happening confidentially. Wait for the announcement of next major version of VS first.

**Reply by zacharylayne** [reply]
I've seen some documentation with instructions to target netstandard2.1. Is this true?

**Reply by huoyaoyuan** [reply]
netstandard2.1 is only supported by a very limited set of platforms. If VS and compilers moves to .NET Core, it should probably use net6.0 or net8.0 as a baseline.

**Reply by CyrusNajmabadi** [reply]
What documentatio nhave you seen that said that @zacharylayne ?

**Reply by zacharylayne** [reply]
@CyrusNajmabadi Apologies for the late reply. I'll try & dig up some of the documentation I saw.

**Reply by WeihanLi** [reply]
Any news for the next major version of VS? Looking forward to the new version

**Reply by 333fred** [reply]
If there is news to share about a new major version of VS, you're not going to hear about it first on this repo 🙂.
